batch.go

```diff
@@ -0,0 +1,126 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+
+	"filippo.io/edwards25519"
+)
+
+// Verifier holds entries of public keys, signature and a scalar which are used for batch verification.
```

Comment:
This is end-user documentation, right?  If so, I think it might be better to have the docs focus on what it does and how, rather than on its internals:
```suggestion
// Verifier performs batch Ed25519 verification using an init-update-finalize API.
```